assets/queries/terraform/alicloud/oss_bucket_allows_all_actions/query.rego

```diff
@@ -0,0 +1,25 @@
+package Cx
+
+import data.generic.common as common_lib
+import data.generic.terraform as terra_lib
+
+CxPolicy[result] {
+
+	json_policy := input.document[i].resource.alicloud_oss_bucket[name].policy
+
+    policy := common_lib.json_unmarshal(json_policy)
+	st := common_lib.get_statement(policy)
+	statement := st[_]
+	statement.Effect == "Allow"
+    terra_lib.anyPrincipal(statement)
+    common_lib.containsOrInArrayContains(statement.Action, "oss:*")
```

Comment:
This code seems to be used in many queries. I suggest creating a function in the library. Something like:

```suggestion
    terra_lib.allows_action_from_all_principals(json_policy, "*")
```

```
allows_action_from_all_principals(json_policy, action) {
  policy := common_lib.json_unmarshal(json_policy)
	st := common_lib.get_statement(policy)
	statement := st[_]
	statement.Effect == "Allow"
    terra_lib.anyPrincipal(statement)
    common_lib.containsOrInArrayContains(statement.Action, action)
}
```

If you agree, you can update the PR #5232 and also the Terraform AWS queries

Reply:
I will do that, thank you :)